### DISCUSSION TITLE
Students interested in contributing to codebase

### DISCUSSION THREAD

**Discussion by durocma** [huridocs/uwazi#5294] [General]
We are students at the University of Michigan taking a class for which the final project involves contributing to an open-source project. 

We were interested in taking a stab at a few of the issues in this codebase and were wondering who to talk to about the project and who to talk to about getting support if need be? 

This project would occur over the next month. How would we go about getting bugs assigned to us? 

Thanks,
Alex and Matt

**Comment by RafaPolit** [comment]
Good morning!  I will be glad to assist.  This is a great contribution, thank you very much.

Let me check a little bit what would be the best place to contribute.  Do you prefer bugs or would it be more interesting to tackle small-ish features that are more interesting to solve?

**Reply by enigmatic-bacon** [reply]
Hi there! I'm Alex, the second partner! 
We had identified the following bugs:
https://github.com/huridocs/uwazi/issues/5172
https://github.com/huridocs/uwazi/issues/5125
https://github.com/huridocs/uwazi/issues/5029
So long as there's an associated open issue, and it involves writing code, it's fair game for our class. 
I've personally interned at Ubisoft and Tesla and enjoyed both writing new features and fixing bugs there. We love the humanitarian aspect of this project and would be open to suggestions if something else is more helpful or a better task for new contributors.

Is this the best place for communication as well? We weren't able to find a community dev Discord or the like.

**Reply by RafaPolit** [reply]
Thanks for the follow up info.  Yes, this is the right place to coordinate and communicate.  Let me check the issues and talk a bit with the team and I'll get back to you.

**Reply by enigmatic-bacon** [reply]
Sounds good!

**Comment by iJustErikk** [comment]
Hi, I am another student at the University of Michigan. I have claimed a few tasks that I did not see any comments on. If there are any issues, please let me know :). Looking forward to contributing :)

These tasks are:
https://github.com/huridocs/uwazi/issues/4963
https://github.com/huridocs/uwazi/issues/5272
https://github.com/huridocs/uwazi/issues/4850

**Comment by RafaPolit** [comment]
@enigmatic-bacon we believe #5172 and #5029 are both good choices.  #5125 will require more input from our side, maybe tackle that last?

@iJustErikk good catch on the #4963 issue, we agree with your comment, and it's OK to tackle that issue.  We believe that #5272 is very complex in a couple of senses: it will require refactoring a big part of the core process of the application, but also that it is not clear for us what the proper solution should be at this point, it probably needs to be part of a larger development we are doing revamping relationships.  Maybe avoid that one? #4850 is grossly mis-documented, but we could provide a bit more feedback.  If you find another one that looks more feasible, please feel free to share here.

**Reply by enigmatic-bacon** [reply]
Hi there Rafael. I've got a running build and have been looking through the codebase throughout the past week. I was wondering if you or another engineer on the team would have time to hop on a call to address a few questions.

I have family in France, so I'm used to the time difference and could work around your schedule. 
Thanks!
Alex

**Reply by konzz** [reply]
Hey @enigmatic-bacon I can join you on a call, contact me and let's schedule something

**Reply by enigmatic-bacon** [reply]
Great! Sent an email. Thanks Alberto.

**Reply by iJustErikk** [reply]
Hi @RafaPolit .  I got around to the small feature for #4963. I tried to push my changes (on a separate branch), but I got met with:

`remote: Permission to huridocs/uwazi.git denied to iJustErikk.` Do I need a developer role? Do you think you could point me to where I could request that? Thank you :)

**Reply by enigmatic-bacon** [reply]
Hi @iJustErikk. You're going to want to fork the project and make a pull request.

**Reply by iJustErikk** [reply]
@enigmatic-bacon thank you! I am very used to the gitlab colab style
